// element-plus/lib/components/popconfirm/src/popconfirm3.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../button/index2.js');
var iconsVue = require('@element-plus/icons-vue');
require('../../../utils/index2.js');
require('../../tooltip/index2.js');
var button = require('../../button/src/button3.js');
var runtime = require('../../../utils/vue/props/runtime2.js');
var icon = require('../../../utils/vue/icon2.js');
var tooltip = require('../../tooltip/src/tooltip3.js');

const popConfirmButtonTypes = [...button.buttonTypes, "text"];
const popconfirmProps = runtime.buildProps({
  title: String,
  confirmButtonText: String,
  cancelButtonText: String,
  confirmButtonType: {
    type: String,
    values: popConfirmButtonTypes,
    default: "primary"
  },
  cancelButtonType: {
    type: String,
    values: popConfirmButtonTypes,
    default: "text"
  },
  icon: {
    type: icon.iconPropType,
    default: iconsVue.QuestionFilled
  },
  iconColor: {
    type: String,
    default: "#f90"
  },
  hideIcon: {
    type: Boolean,
    default: false
  },
  hideAfter: {
    type: Number,
    default: 200
  },
  onConfirm: {
    type: runtime.definePropType(Function)
  },
  onCancel: {
    type: runtime.definePropType(Function)
  },
  teleported: tooltip.useTooltipContentProps.teleported,
  persistent: tooltip.useTooltipContentProps.persistent
});

exports.popconfirmProps = popconfirmProps;
