// element-plus/lib/components/checkbox/index2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../utils/index2.js');
var checkbox$1 = require('./src/checkbox4.js');
var checkboxButton = require('./src/checkbox-button2.js');
var checkboxGroup = require('./src/checkbox-group2.js');
var checkbox = require('./src/checkbox3.js');
require('./src/checkbox.type2.js');
var install = require('../../utils/vue/install2.js');

const ElCheckbox = install.withInstall(checkbox$1["default"], {
  CheckboxButton: checkboxButton["default"],
  CheckboxGroup: checkboxGroup["default"]
});
const ElCheckboxButton = install.withNoopInstall(checkboxButton["default"]);
const ElCheckboxGroup = install.withNoopInstall(checkboxGroup["default"]);

exports.checkboxEmits = checkbox.checkboxEmits;
exports.checkboxGroupEmits = checkbox.checkboxGroupEmits;
exports.checkboxProps = checkbox.checkboxProps;
exports.useCheckbox = checkbox.useCheckbox;
exports.useCheckboxGroup = checkbox.useCheckboxGroup;
exports.useCheckboxGroupId = checkbox.useCheckboxGroupId;
exports.useCheckboxGroupProps = checkbox.useCheckboxGroupProps;
exports.ElCheckbox = ElCheckbox;
exports.ElCheckboxButton = ElCheckboxButton;
exports.ElCheckboxGroup = ElCheckboxGroup;
exports["default"] = ElCheckbox;
