// element-plus/lib/components/tooltip/src/tooltip3.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index2.js');
require('../../popper/index2.js');
require('../../../hooks/index2.js');
require('../../../constants/index2.js');
var index = require('../../../hooks/use-namespace/index2.js');
var runtime = require('../../../utils/vue/props/runtime2.js');
var index$1 = require('../../../hooks/use-delayed-toggle/index2.js');
var content = require('../../popper/src/content3.js');
var index$2 = require('../../../hooks/use-popper-container/index2.js');
var trigger = require('../../popper/src/trigger3.js');
var aria = require('../../../constants/aria2.js');

const triggers = ["hover", "focus", "click", "contextmenu"];
const ns = index.useNamespace("tooltip");
const useTooltipContentProps = runtime.buildProps({
  ...index$1.useDelayedToggleProps,
  ...content.usePopperContentProps,
  appendTo: {
    type: runtime.definePropType([String, Object]),
    default: index$2.POPPER_CONTAINER_SELECTOR
  },
  content: {
    type: String,
    default: ""
  },
  rawContent: {
    type: Boolean,
    default: false
  },
  persistent: Boolean,
  ariaLabel: String,
  visible: {
    type: runtime.definePropType(Boolean),
    default: null
  },
  transition: {
    type: String,
    default: `${ns.namespace.value}-fade-in-linear`
  },
  teleported: {
    type: Boolean,
    default: true
  },
  disabled: {
    type: Boolean
  }
});
const useTooltipTriggerProps = runtime.buildProps({
  ...trigger.usePopperTriggerProps,
  disabled: Boolean,
  trigger: {
    type: runtime.definePropType([String, Array]),
    default: "hover"
  },
  triggerKeys: {
    type: runtime.definePropType(Array),
    default: () => [aria.EVENT_CODE.enter, aria.EVENT_CODE.space]
  }
});
const useTooltipProps = runtime.buildProps({
  openDelay: {
    type: Number
  },
  visibleArrow: {
    type: Boolean,
    default: void 0
  },
  hideAfter: {
    type: Number,
    default: 200
  },
  showArrow: {
    type: Boolean,
    default: true
  }
});

exports.useTooltipContentProps = useTooltipContentProps;
exports.useTooltipProps = useTooltipProps;
exports.useTooltipTriggerProps = useTooltipTriggerProps;
