// element-plus/lib/components/radio/src/radio3.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index2.js');
require('../../../constants/index2.js');
require('../../../hooks/index2.js');
var runtime = require('../../../utils/vue/props/runtime2.js');
var index = require('../../../hooks/use-common-props/index2.js');
var event = require('../../../constants/event2.js');
var shared = require('@vue/shared');
var core = require('@vueuse/core');

const radioPropsBase = runtime.buildProps({
  size: index.useSizeProp,
  disabled: Boolean,
  label: {
    type: [String, Number, Boolean],
    default: ""
  }
});
const radioProps = runtime.buildProps({
  ...radioPropsBase,
  modelValue: {
    type: [String, Number, Boolean],
    default: ""
  },
  name: {
    type: String,
    default: ""
  },
  border: Boolean
});
const radioEmits = {
  [event.UPDATE_MODEL_EVENT]: (val) => shared.isString(val) || core.isNumber(val) || core.isBoolean(val),
  [event.CHANGE_EVENT]: (val) => shared.isString(val) || core.isNumber(val) || core.isBoolean(val)
};

exports.radioEmits = radioEmits;
exports.radioProps = radioProps;
exports.radioPropsBase = radioPropsBase;
