// element-plus/lib/components/message/src/message3.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index2.js');
var runtime = require('../../../utils/vue/props/runtime2.js');
var icon = require('../../../utils/vue/icon2.js');

const messageTypes = ["success", "info", "warning", "error"];
const messageProps = runtime.buildProps({
  customClass: {
    type: String,
    default: ""
  },
  center: {
    type: Boolean,
    default: false
  },
  dangerouslyUseHTMLString: {
    type: Boolean,
    default: false
  },
  duration: {
    type: Number,
    default: 3e3
  },
  icon: {
    type: icon.iconPropType,
    default: ""
  },
  id: {
    type: String,
    default: ""
  },
  message: {
    type: runtime.definePropType([
      String,
      Object,
      Function
    ]),
    default: ""
  },
  onClose: {
    type: runtime.definePropType(Function),
    required: false
  },
  showClose: {
    type: Boolean,
    default: false
  },
  type: {
    type: String,
    values: messageTypes,
    default: "info"
  },
  offset: {
    type: Number,
    default: 20
  },
  zIndex: {
    type: Number,
    default: 0
  },
  grouping: {
    type: Boolean,
    default: false
  },
  repeatNum: {
    type: Number,
    default: 1
  }
});
const messageEmits = {
  destroy: () => true
};

exports.messageEmits = messageEmits;
exports.messageProps = messageProps;
exports.messageTypes = messageTypes;
