// element-plus/es/components/radio/src/radio3.mjs
import '../../../utils/index2.mjs';
import '../../../constants/index2.mjs';
import '../../../hooks/index2.mjs';
import { buildProps } from '../../../utils/vue/props/runtime2.mjs';
import { useSizeProp } from '../../../hooks/use-common-props/index2.mjs';
import { UPDATE_MODEL_EVENT, CHANGE_EVENT } from '../../../constants/event2.mjs';
import { isString } from '@vue/shared';
import { isNumber, isBoolean } from '@vueuse/core';

const radioPropsBase = buildProps({
  size: useSizeProp,
  disabled: Boolean,
  label: {
    type: [String, Number, Boolean],
    default: ""
  }
});
const radioProps = buildProps({
  ...radioPropsBase,
  modelValue: {
    type: [String, Number, Boolean],
    default: ""
  },
  name: {
    type: String,
    default: ""
  },
  border: Boolean
});
const radioEmits = {
  [UPDATE_MODEL_EVENT]: (val) => isString(val) || isNumber(val) || isBoolean(val),
  [CHANGE_EVENT]: (val) => isString(val) || isNumber(val) || isBoolean(val)
};

export { radioEmits, radioProps, radioPropsBase };
